src/agni/server/receiver/UserReceiver.java

```diff
@@ -1,17 +1,55 @@
 package agni.server.receiver;
 
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Vector;
+
 public class UserReceiver implements MessageParser {
-    private UserListener[] userListeners;
+    private Vector <UserListener> userListeners = null;
 
     public UserReceiver() {
-
+        userListeners = new Vector<UserListener>();
     }
 
-    private void notifyUserRequest() {
+    private void notifyUserRequest(String ip, byte type, String message) {
```

Comment:
Since the type is parsed and immediately passed I don't think it makes sense to translate it to an enum then immediately translate it back to bytes for passing. I think it will just complicate the code.